src/components/Form/formFunctions.ts

```diff
@@ -58,12 +58,8 @@ export const createSetter = <SpecificFields extends Fields>(
     setFields: (SpecificFields: SpecificFields) => void,
     fieldValues: SpecificFields
 ): Setter<SpecificFields> => {
-    return <Key extends keyof SpecificFields>(keyValuePairs: { key: Key; value: any }[]): void => {
-        const newFieldValues: SpecificFields = { ...fieldValues };
-        keyValuePairs.forEach((keyValuePair) => {
-            newFieldValues[keyValuePair.key] = keyValuePair.value;
-        });
-        setFields(newFieldValues);
+    return (fieldValuesToUpdate: Partial<SpecificFields>): void => {
+        setFields({ ...fieldValues, ...fieldValuesToUpdate });
```

Comment:
I thought the same when I first saw this, but it does retain all updates. I guess even though fieldValues is passed in when creating the setter, it's by reference instead of value or something? 
I've just double checked and it definitely retains the updates :)

Reply:
ok cool!